### PR TITLE
Fix double import

### PR DESCRIPTION
**Pull request by camilamacedo86**
**Description**
- Add a new unit test
- Fix issue where webhooks were to checking/validating the input data. 

**Motivation**
https://github.com/kubernetes-sigs/kubebuilder/issues/1069

### CODE REVIEW

**Comment by camilamacedo86** [comment]
HI @mengqiy. 

Could you help us by doing the review of the issue faced with the golden script (double import after the introduction of GroupImportSafe)?

File: pkg/scaffold/v1/resource/resource_test.go [hunk 1]
```diff
@@ -119,6 +119,12 @@ var _ = Describe("Resource", func() {
 			Expect(instance.Resource).To(Equal("cats"))
 		})
 
+		It("should allow webapp-project as a Group", func() {
```
Comment (DirectXMan12) [inline pkg/scaffold/v1/resource/resource_test.go]: test names should describe the intent of the test, so this should say something like "should allow hyphens in group names".

Comment (camilamacedo86) [reply pkg/scaffold/v1/resource/resource_test.go]: Done 👍 Tks for your input.

**Comment by camilamacedo86** [comment]
Hi @DirectXMan12,

Could we merge this one and the https://github.com/kubernetes-sigs/kubebuilder/pull/1054 which is blocked by it now?

**Pull request merged**